tasks/artifact.xml

```diff
@@ -0,0 +1,314 @@
+        <exec dir="${artifact.directory}" command="git branch | grep ${artifact.git.temporary_branch} &amp;&amp; git branch -D ${artifact.git.temporary_branch}" logoutput="true" />
+        <exec dir="${artifact.directory}" command="git branch ${artifact.git.temporary_branch} ${artifact.git.remote_name}/${artifact.git.remote_branch}" checkreturn="true" logoutput="true" />
+        <exec dir="${artifact.directory}" command="git checkout ${artifact.git.temporary_branch}" checkreturn="true" />
+        <echo>Building on temporary branch '${artifact.git.temporary_branch}'.</echo>
+    </target>
+
+
+    <target name="artifact-updateCode">
+        <!-- Remove all the existing files so that we can install cleanly. -->
+        <delete includeemptydirs="true">
+            <fileset dir="${artifact.directory}" defaultexcludes="false" excludes=".git,.git/**" includes="**/*,**/.git,**/.git/**" />
+        </delete>
+
+        <!-- List all files that are checked in to git, then use the list to copy them all
+             into the artifact. -->
+        <tempfile property="tmpfile" destdir="${build.dir}/artifacts" />
+        <exec command="git ls-files" dir="${build.dir}" output="${tmpfile}" />
+
+        <copy todir="${artifact.directory}" preservemode="false" mode="750" overwrite="true" haltonerror="true">
+            <filelist dir="${build.dir}" listfile="${tmpfile}" />
+        </copy>
+        <delete file="${tmpfile}" />
+
+        <!-- Copy a template .gitignore specific to the artifact. -->
+        <copy file="${artifact.gitignore_template}" tofile="${artifact.directory}/.gitignore" overwrite="true">
+            <filterchain>
+                <expandproperties />
+            </filterchain>
+        </copy>
+
+        <!-- Copy a template README into the artifact. -->
+        <copy file="${artifact.readme_template}" tofile="${artifact.directory}/README.md" overwrite="true">
+            <filterchain>
+                <expandproperties />
+            </filterchain>
+        </copy>
+    </target>
+
+
+    <target name="artifact-build">
+        <echo>Installing composer dependencies in the artifact...</echo>
+        <exec dir="${artifact.directory}" command="composer install --no-interaction --no-dev" checkreturn="true" />
+
+        <echo>Deleting .git subdirectories added by Composer...</echo>
+        <delete includeemptydirs="true">
+            <fileset dir="${artifact.directory}" defaultexcludes="false" excludes=".gitignore,.git/,.git/**" includes="**/.gitignore,**/.git,**/.git/**" />
+        </delete>
+
+        <echo>Running 'phing build' in the artifact...</echo>
+        <!--
+            - Override the Drupal root so that it is within the artifact directory.
+            - Override the "artifact mode" so that files managed with the
+              <includeresource /> task are copied into the artifact instead of symlinked.
+            -->
+        <phing target="build" phingfile="${phing.file}" inheritAll="false" dir=".">
```

Comment:
Would `inheritAll` keep `build.env` from being passed here?  When I tested this, the artifact was built with a `settings.php` for The Vagrant instead of Acquia's.

Reply:
Never mind.  I didn't have an `build.acquia.properties` that specified the Acquia settings file.